main/src/library/Float32.flix

```diff
@@ -104,4 +110,91 @@ namespace Float32 {
      } catch {
          case _: ##java.lang.NumberFormatException => Err("Float32.fromString")
      }
+
+    ///
+    /// Convert `x` to an `Option[Int8]`.
+    ///
+    /// If the numeric value of `x` is within the range of Int8 return `Some(x as Int8)`, rounding `x` towards 0`.
+    ///
+    /// If the numeric value of `x` is outside the range of Int8, or it is NaN or infinity, return `None`.
+    ///
+    pub def toInt8(x: Float32): Option[Int8] =
+        import java.lang.Float:valueOf(Float32);
+        import java.lang.Float.byteValue();
+        if (x < Int8.toFloat32(Int8.minValue()) || x > Int8.toFloat32(Int8.maxValue()) || isNan(x) || isInfinite(x))
```

Comment:
OK; I think I understand this.

I was wondering if we should have some operation where you can supply a value for the "negative infinity" and "positive infinity" cases and/or where they are automatically "clamped" to Int8.minValue() and Int8.maxValue(). Kind of like how we have `Option.getWithDefault`.

Reply:
Maybe like `toInt8WithMinMax` and/or `toInt8With(min = ..., max = ...)`

Reply:
I like the idea of supplying a min and max value for clamping, because the min and max values will be in the target type you won't be able to mix up the arg order if you you thinking of UFCS or "not UFCS". 

I think I'd prefer to have "clamp" in the name though, rather than say `toInt8WithMinMax`,  this would stop flag to a new user that the clamp version shouldn't be seen as the default choice for a conversion function.

Because we have `Option.getWithDefault` in UFCS order I'll add functions like this:

~~~

pub def clampToInt8(x: Int64, min: Int8, max: Int8): Int8 = 

~~~

Reply:
Great!

Reply:
I've push the clamp functions but the tests are either incomplete or missing.